plugins/webhooks/helper.js

```diff
@@ -0,0 +1,1192 @@
+ * @param    {String}     ref              The reference which we want
+ * @param    {String}     checkoutUrl      Scm checkout URL
+ * @param    {String}     scmContext       Scm which pipeline's repository exists in
+ * @returns  {Promise}                     Specific SHA1 commit to start the build with
+ */
+async function getCommitRefSha({ scm, token, ref, refType, checkoutUrl, scmContext }) {
+    // For example, git@github.com:screwdriver-cd/data-schema.git => screwdriver-cd, data-schema
+    const owner = CHECKOUT_URL_SCHEMA_REGEXP.exec(checkoutUrl)[2];
+    const repo = CHECKOUT_URL_SCHEMA_REGEXP.exec(checkoutUrl)[3];
+
+    return scm.getCommitRefSha({
+        token,
+        owner,
+        repo,
+        ref,
+        refType,
+        scmContext
+    });
+}
+
+/**
+ * Start pipeline events from scm webhook config
+ * @method startHookEvent
+ * @param  {Hapi.request} request   Request from user
+ * @param  {Object} h               Response toolkit
+ * @param  {Object} webhookConfig   Parsed request.payload by scm webhook
+ * @return {Promise}
+ */
+async function startHookEvent(request, h, webhookConfig) {
+    const { userFactory, pipelineFactory } = request.server.app;
+    const { scm } = pipelineFactory;
+    const ignoreUser = webhookConfig.pluginOptions.ignoreCommitsBy;
+    let message = 'Unable to process this kind of event';
+    let skipMessage;
+    let parsedHookId = '';
+
+    const { type, hookId, username, scmContext, ref, checkoutUrl, action, prNum } = webhookConfig;
+
+    parsedHookId = hookId;
+
+    try {
+        // skipping checks
+        if (/\[(skip ci|ci skip)\]/.test(webhookConfig.lastCommitMessage)) {
+            skipMessage = 'Skipping due to the commit message: [skip ci]';
+        }
+
+        // if skip ci then don't return
+        if (ignoreUser && ignoreUser.length !== 0 && !skipMessage) {
+            const commitAuthors =
+                Array.isArray(webhookConfig.commitAuthors) && webhookConfig.commitAuthors.length !== 0
+                    ? webhookConfig.commitAuthors
+                    : [username];
+            const validCommitAuthors = commitAuthors.filter(author => !ignoreUser.includes(author));
+
+            if (!validCommitAuthors.length) {
+                message = `Skipping because user ${username} is ignored`;
+                request.log(['webhook', hookId], message);
+
+                return h.response({ message }).code(204);
+            }
+        }
+
+        const token = await obtainScmToken({ pluginOptions: webhookConfig.pluginOptions, userFactory, username, scmContext, scm });
+
+        if (action !== 'release' && action !== 'tag') {
+            let scmUri;
+
+            if (type === 'pr') {
+                scmUri = await scm.parseUrl({ checkoutUrl, token, scmContext });
+            }
+            webhookConfig.changedFiles = await scm.getChangedFiles({
+                webhookConfig,
```

Comment:
this might not be able to omit because the name is different.
```suggestion
                payload: webhookConfig,
```

Reply:
and does `webhookConfig` contain `request.payload`?

Reply:
The argument of `getChangedFiles` will turn into `webhookConfig` at https://github.com/screwdriver-cd/scm-github/pull/200/files#diff-e727e4bdf3657fd1d798edcd6b099d6e092f8573cba266154583a746bba0f346R1318.

Reply:
make sense